---
title: "ASC TLR10 Proteomics - Differential Expression Analysis"
output:
  html_document:
    df_print: paged
    toc: true
    toc_float: true
    toc_depth: 4
---

# Setup

```{r setup, message=FALSE, warning=FALSE}

source("helpers.R")

dir.create("results/TLR4/DEA", showWarnings = F, recursive = T)

knitr::opts_chunk$set(fig.width = 10, dpi = 300, results = "hold", fig.show = "hold")

```

```{r plot_settings}

# Heatmaps

hm_cluster_rows <- TRUE # Genes
hm_cluster_cols <- TRUE # Samples
hm_scale_by_row <- TRUE
hm_max_rows <- 30

heatmap.colors <- colorRamp2(c(-2, 0, 2), c("darkblue", "white", "darkred"))

# Volcano plot

vp_lfc_limit <- 5

```

# Undifferentiated

## Supernatant

```{r ASC_SN_import}

data <- read_DIA_report("data/TLR4/20240828_120228_ASC_TERT_TLR_10_4_SN_Report.tsv")

data <- data[, data$Cell.Type != "ASC_TERT_TLR10_LOV"]

data_MS1.df <- assay(data, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
data_MS2.df <- assay(data, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")  

```

```{r ASC_SN_log_transform}

data.log2 <- data
assays(data.log2) %<>% lapply(log2)

data_MS1.log2.df <- assay(data.log2, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.log2.df <- assay(data.log2, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")


plot_intensity_boxplot(data_MS1.log2.df, data_MS2.log2.df)

```

```{r ASC_SN_normalize}

data.norm <- data.log2
assays(data.norm) %<>% lapply(normalize_matrix, "center.median")

data_MS1.norm.df <- assay(data.norm, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.norm.df <- assay(data.norm, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

plot_intensity_boxplot(data_MS1.norm.df, data_MS2.norm.df)

```

```{r ASC_SN_plot_PCA}

plot_pca(data.norm)

```

```{r ASC_SN_fit_model}


contrast_list <- c(
  LightVsDark = "ASC_TERT_TLR4_LOV.Light - ASC_TERT_TLR4_LOV.Dark",
  LPSVsDark = "ASC_TERT_TLR4_LOV.LPS - ASC_TERT_TLR4_LOV.Dark",
  LightVsLPS = "ASC_TERT_TLR4_LOV.Light - ASC_TERT_TLR4_LOV.LPS"
)

fit <- fit_DEqMS_model(data.norm, contrast_list)

VarianceBoxplot(fit)

```

### ASC TERT TLR4LOV Light vs. Dark

```{r TLR4_Light_Dark_SN_extract_results}

current_contrast <- 1
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.norm[rownames(data.norm) %in% res.sig$gene, 
                         str_detect(colnames(data.norm), "Light|Dark")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/TLR4/DEA/ASC_TERT_TLR4LOV_Light_vs_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/TLR4/DEA/ASC_TERT_TLR4LOV_Light_vs_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r TLR4_Light_Dark_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r TLR4_Light_Dark_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_TERT_TLR4_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r TLR4_Light_Dark_volcanoplot}

plot_volcano(res, "ASC TERT TLR4LOV Light vs. Dark", vp_lfc_limit)

```


### ASC TLR10LOV Light vs. Dark

```{r ASC_TLR10_SN_extract_results}

current_contrast <- 2
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "TLR10_LOV")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_TLR10LOV_Light_vs_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_TLR10LOV_Light_vs_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r ASC_TLR10_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_TLR10_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_TLR10_SN_volcanoplot}

plot_volcano(res, "ASC TLR10LOV Light vs. Dark Supernatant", vp_lfc_limit)

```


### ASC TLR10LOV vs. Wildtype Dark

```{r ASC_Dark_SN_extract_results}

current_contrast <- 3
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Dark")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_WT_vs_TLR10LOV_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_WT_vs_TLR10LOV_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r ASC_Dark_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_Dark_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_Dark_SN_volcanoplot}

plot_volcano(res, "ASC TLR10LOV vs. Wildtype Dark Supernatant", vp_lfc_limit)

```


### ASC TLR10LOV vs. Wildtype Light

```{r ASC_Light_SN_extract_results}

current_contrast <- 4
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Light")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_WT_vs_TLR10LOV_Light_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_WT_vs_TLR10LOV_Light_SN_filtered.csv",
          row.names = FALSE)

```

```{r ASC_Light_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_Light_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows,
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_Light_SN_volcanoplot}

plot_volcano(res, "ASC TLR10LOV vs. Wildtype Light Supernatant", vp_lfc_limit)

```


### ASC Interaction (TLR10LOV Light vs. Dark) vs. (Wildtype Light vs. Dark)

```{r ASC_Interaction_SN_extract_results}

current_contrast <- 5
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, ]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_Interaction_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_Interaction_SN_filtered.csv",
          row.names = FALSE)

```

```{r ASC_Interaction_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_Interaction_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, cluster_rows = hm_cluster_rows,
             cluster_cols = hm_cluster_cols, scale_by_row = hm_scale_by_row)

```

```{r ASC_Interaction_SN_volcanoplot}

plot_volcano(res, "ASC Interaction Supernatant", vp_lfc_limit)

```


## Whole Cell Lysate

```{r ASC_WCL_import}

data <- read_DIA_report("data/MS1MS2/20240701_082255_ASC_WCL_TLR10_Light_Report.tsv")

data_MS1.df <- assay(data, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
data_MS2.df <- assay(data, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```

```{r ASC_WCL_plot_missing_before}

plot_missing(data_MS1.df, data_MS2.df)

```

```{r ASC_WCL_filter_na}

keep <- filter_too_many_missing(data_MS1.df, data_MS2.df)

data.filtered <- data[rownames(data) %in% keep,]


data_MS1.filtered.df <- assay(data.filtered, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.filtered.df <- assay(data.filtered, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```


```{r ASC_WCL_plot_missing_after}

plot_missing(data_MS1.filtered.df, data_MS2.filtered.df)

```

```{r ASC_WCL_log_transform}

data.log2 <- data.filtered
assays(data.log2) %<>% lapply(log2)

data_MS1.log2.df <- assay(data.log2, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.log2.df <- assay(data.log2, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")


plot_intensity_boxplot(data_MS1.log2.df, data_MS2.log2.df)

```

```{r ASC_WCL_normalize}

data.norm <- data.log2
assays(data.norm) %<>% lapply(normalize_matrix, "center.median")

data_MS1.norm.df <- assay(data.norm, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.norm.df <- assay(data.norm, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

plot_intensity_boxplot(data_MS1.norm.df, data_MS2.norm.df)

```

```{r ASC_WCL_plot_missing_heatmap}

MNAR_MS1 <- get_MNAR(data_MS1.norm.df)
MNAR_MS2 <- get_MNAR(data_MS2.norm.df)
MNAR <- MNAR_MS1 | MNAR_MS2

plot_missing_heatmap(data_MS1.norm.df, data_MS2.norm.df, MNAR, colData(data.norm),
                     colors_columns = list(Celltype=c("ASC_WT" = "lightblue",
                                                      "ASC_TLR10_LOV" = "lightgreen"),
                                           Condition=c("Light" = "orange",
                                                       "Dark" = "midnightblue")),
                     colors_rows = list(MNAR=c("TRUE"="darkgreen",
                                               "FALSE"="darkmagenta")))

```

```{r ASC_WCL_plot_missing_density}

plot_missing_density(data_MS1.norm.df, data_MS2.norm.df)

```

```{r ASC_WCL_impute_missing_values}

data.imputed <- data.norm

assays(data.imputed) %<>% lapply(MsCoreUtils::impute_mixed, !MNAR, "knn", "min")

data_MS1.imputed.df <- assay(data.imputed, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.imputed.df <- assay(data.imputed, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
  
```

```{r ASC_WCL_plot_PCA}

plot_pca(data.imputed, scale = F)

```

```{r ASC_WCL_fit_model}


contrast_list <- c(
  LightVsDarkInWT = "ASC_WT.Light - ASC_WT.Dark", 
  LightVsDarkInTLR10 = "ASC_TLR10_LOV.Light - ASC_TLR10_LOV.Dark",
  TLR10VsWTInDark = "ASC_TLR10_LOV.Dark - ASC_WT.Dark",
  TLR10VsWTInLight = "ASC_TLR10_LOV.Light - ASC_WT.Light",
  Diff = "(ASC_TLR10_LOV.Light - ASC_TLR10_LOV.Dark) - (ASC_WT.Light - ASC_WT.Dark)"
)

fit <- fit_DEqMS_model(data.imputed, contrast_list)

VarianceBoxplot(fit)

```

### ASC Wildtype Light vs. Dark

```{r ASC_WT_WCL_extract_results}

current_contrast <- 1
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "WT")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_WT_Light_vs_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_WT_Light_vs_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ASC_WT_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_WT_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_WT_WCL_volcanoplot}

plot_volcano(res, "ASC Wildtype Light vs. Dark Whole Cell Lysate", vp_lfc_limit)

```


### ASC TLR10LOV Light vs. Dark

```{r ASC_TLR10_WCL_extract_results}

current_contrast <- 2
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "TLR10_LOV")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_TLR10LOV_Light_vs_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_TLR10LOV_Light_vs_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ASC_TLR10_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_TLR10_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_TLR10_WCL_volcanoplot}

plot_volcano(res, "ASC TLR10LOV Light vs. Dark Whole Cell Lysate", vp_lfc_limit)

```


### ASC TLR10LOV vs. Wildtype Dark

```{r ASC_Dark_WCL_extract_results}

current_contrast <- 3
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Dark")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_WT_vs_TLR10LOV_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_WT_vs_TLR10LOV_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ASC_Dark_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_Dark_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_Dark_WCL_volcanoplot}

plot_volcano(res, "ASC TLR10LOV vs. Wildtype Dark Whole Cell Lysate", vp_lfc_limit)

```


### ASC TLR10LOV vs. Wildtype Light

```{r ASC_Light_WCL_extract_results}

current_contrast <- 4
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Light")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_WT_vs_TLR10LOV_Light_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_WT_vs_TLR10LOV_Light_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ASC_Light_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_Light_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows,
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_Light_WCL_volcanoplot}

plot_volcano(res, "ASC TLR10LOV vs. Wildtype Light Whole Cell Lysate", vp_lfc_limit)

```


### ASC Interaction TLR10LOV Light vs. Dark vs. Wildtype Light vs. Dark

```{r ASC_Interaction_WCL_extract_results}

current_contrast <- 5
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, ]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ASC_Interaction_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ASC_Interaction_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ASC_Interaction_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ASC_Interaction_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ASC_WT" = "lightblue",
                                                       "ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ASC_Interaction_WCL_volcanoplot}

plot_volcano(res, "ASC Interaction Whole Cell Lysate", vp_lfc_limit)

```


# ADIPO ASC

## Supernatant

```{r ADIPO_ASC_SN_import}

data <- read_DIA_report("data/MS1MS2/20240701_082311_ADIPO_ASC_SN_TLR10_Light_Report.tsv")

data_MS1.df <- assay(data, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
data_MS2.df <- assay(data, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```

```{r ADIPO_ASC_SN_plot_missing_before}

plot_missing(data_MS1.df, data_MS2.df)

```

```{r ADIPO_ASC_SN_filter_na}

keep <- filter_too_many_missing(data_MS1.df, data_MS2.df)

data.filtered <- data[rownames(data) %in% keep,]


data_MS1.filtered.df <- assay(data.filtered, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.filtered.df <- assay(data.filtered, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```


```{r ADIPO_ASC_SN_plot_missing_after}

plot_missing(data_MS1.filtered.df, data_MS2.filtered.df)

```

```{r ADIPO_ASC_SN_log_transform}

data.log2 <- data.filtered
assays(data.log2) %<>% lapply(log2)

data_MS1.log2.df <- assay(data.log2, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.log2.df <- assay(data.log2, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")


plot_intensity_boxplot(data_MS1.log2.df, data_MS2.log2.df)

```

```{r ADIPO_ASC_SN_normalize}

data.norm <- data.log2
assays(data.norm) %<>% lapply(normalize_matrix, "center.median")

data_MS1.norm.df <- assay(data.norm, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.norm.df <- assay(data.norm, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

plot_intensity_boxplot(data_MS1.norm.df, data_MS2.norm.df)

```

```{r ADIPO_ASC_SN_plot_missing_heatmap}

MNAR_MS1 <- get_MNAR(data_MS1.norm.df)
MNAR_MS2 <- get_MNAR(data_MS2.norm.df)
MNAR <- MNAR_MS1 | MNAR_MS2

plot_missing_heatmap(data_MS1.norm.df, data_MS2.norm.df, MNAR, colData(data.norm),
                     colors_columns = list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                      "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                           Condition=c("Light" = "orange",
                                                       "Dark" = "midnightblue")),
                     colors_rows = list(MNAR=c("TRUE"="darkgreen",
                                               "FALSE"="darkmagenta")))

```

```{r ADIPO_ASC_SN_plot_missing_density}

plot_missing_density(data_MS1.norm.df, data_MS2.norm.df)

```

```{r ADIPO_ASC_SN_impute_missing_values}

data.imputed <- data.norm

assays(data.imputed) %<>% lapply(MsCoreUtils::impute_mixed, !MNAR, "knn", "min")

data_MS1.imputed.df <- assay(data.imputed, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.imputed.df <- assay(data.imputed, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
  
```

```{r ADIPO_ASC_SN_plot_PCA}

plot_pca(data.imputed, scale = F)

```

```{r ADIPO_ASC_SN_fit_model}


contrast_list <- c(
  LightVsDarkInWT = "ADIPO_ASC_WT.Light - ADIPO_ASC_WT.Dark", 
  LightVsDarkInTLR10 = "ADIPO_ASC_TLR10_LOV.Light - ADIPO_ASC_TLR10_LOV.Dark",
  TLR10VsWTInDark = "ADIPO_ASC_TLR10_LOV.Dark - ADIPO_ASC_WT.Dark",
  TLR10VsWTInLight = "ADIPO_ASC_TLR10_LOV.Light - ADIPO_ASC_WT.Light",
  Diff = "(ADIPO_ASC_TLR10_LOV.Light - ADIPO_ASC_TLR10_LOV.Dark) - (ADIPO_ASC_WT.Light - ADIPO_ASC_WT.Dark)"
)

fit <- fit_DEqMS_model(data.imputed, contrast_list)

VarianceBoxplot(fit)

```

### ADIPO ASC Wildtype Light vs. Dark

```{r ADIPO_ASC_WT_SN_extract_results}

current_contrast <- 1
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "WT")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_WT_Light_vs_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_WT_Light_vs_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_WT_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_WT_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_WT_SN_volcanoplot}

plot_volcano(res, "ADIPO ASC Wildtype Light vs. Dark Supernatant", vp_lfc_limit)

```


### ADIPO ASC TLR10LOV Light vs. Dark

```{r ADIPO_ASC_TLR10_SN_extract_results}

current_contrast <- 2
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "TLR10_LOV")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_TLR10LOV_Light_vs_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_TLR10LOV_Light_vs_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_TLR10_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_TLR10_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_TLR10_SN_volcanoplot}

plot_volcano(res, "ADIPO ASC TLR10LOV Light vs. Dark Supernatant", vp_lfc_limit)

```


### ADIPO ASC TLR10LOV vs. Wildtype Dark

```{r ADIPO_ASC_Dark_SN_extract_results}

current_contrast <- 3
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Dark")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_Dark_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_Dark_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_Dark_SN_volcanoplot}

plot_volcano(res, "ADIPO ASC TLR10LOV vs. Wildtype Dark Supernatant", vp_lfc_limit)

```


### ADIPO ASC TLR10LOV vs. Wildtype Light

```{r ADIPO_ASC_Light_SN_extract_results}

current_contrast <- 4
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Light")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Light_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Light_SN_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_Light_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_Light_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows,
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_Light_SN_volcanoplot}

plot_volcano(res, "ADIPO ASC TLR10LOV vs. Wildtype Light Supernatant", vp_lfc_limit)

```


### ADIPO ASC Interaction (TLR10LOV Light vs. Dark) vs. (Wildtype Light vs. Dark)

```{r ADIPO_ASC_Interaction_SN_extract_results}

current_contrast <- 5
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, ]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_Interaction_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_Interaction_SN_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_Interaction_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_Interaction_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, cluster_rows = hm_cluster_rows,
             cluster_cols = hm_cluster_cols, scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_Interaction_SN_volcanoplot}

plot_volcano(res, "ADIPO ASC Interaction Supernatant", vp_lfc_limit)

```


## Whole Cell Lysate

```{r ADIPO_ASC_WCL_import}

data <- read_DIA_report("data/MS1MS2/20240626_081714_ADIPO_ASC_WCL_TLR10_Light_Report.tsv")

data_MS1.df <- assay(data, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
data_MS2.df <- assay(data, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```

```{r ADIPO_ASC_WCL_plot_missing_before}

plot_missing(data_MS1.df, data_MS2.df)

```

```{r ADIPO_ASC_WCL_filter_na}

keep <- filter_too_many_missing(data_MS1.df, data_MS2.df)

data.filtered <- data[rownames(data) %in% keep,]


data_MS1.filtered.df <- assay(data.filtered, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.filtered.df <- assay(data.filtered, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```


```{r ADIPO_ASC_WCL_plot_missing_after}

plot_missing(data_MS1.filtered.df, data_MS2.filtered.df)

```

```{r ADIPO_ASC_WCL_log_transform}

data.log2 <- data.filtered
assays(data.log2) %<>% lapply(log2)

data_MS1.log2.df <- assay(data.log2, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.log2.df <- assay(data.log2, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")


plot_intensity_boxplot(data_MS1.log2.df, data_MS2.log2.df)

```

```{r ADIPO_ASC_WCL_normalize}

data.norm <- data.log2
assays(data.norm) %<>% lapply(normalize_matrix, "center.median")

data_MS1.norm.df <- assay(data.norm, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.norm.df <- assay(data.norm, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

plot_intensity_boxplot(data_MS1.norm.df, data_MS2.norm.df)

```

```{r ADIPO_ASC_WCL_plot_missing_heatmap}

MNAR_MS1 <- get_MNAR(data_MS1.norm.df)
MNAR_MS2 <- get_MNAR(data_MS2.norm.df)
MNAR <- MNAR_MS1 | MNAR_MS2

plot_missing_heatmap(data_MS1.norm.df, data_MS2.norm.df, MNAR, colData(data.norm),
                     colors_columns = list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                      "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                           Condition=c("Light" = "orange",
                                                       "Dark" = "midnightblue")),
                     colors_rows = list(MNAR=c("TRUE"="darkgreen",
                                               "FALSE"="darkmagenta")))

```

```{r ADIPO_ASC_WCL_plot_missing_density}

plot_missing_density(data_MS1.norm.df, data_MS2.norm.df)

```

```{r ADIPO_ASC_WCL_impute_missing_values}

data.imputed <- data.norm

assays(data.imputed) %<>% lapply(MsCoreUtils::impute_mixed, !MNAR, "knn", "min")

data_MS1.imputed.df <- assay(data.imputed, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.imputed.df <- assay(data.imputed, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
  
```

```{r ADIPO_ASC_WCL_plot_PCA}

plot_pca(data.imputed, scale = F)

```

```{r ADIPO_ASC_WCL_fit_model}


contrast_list <- c(
  LightVsDarkInWT = "ADIPO_ASC_WT.Light - ADIPO_ASC_WT.Dark", 
  LightVsDarkInTLR10 = "ADIPO_ASC_TLR10_LOV.Light - ADIPO_ASC_TLR10_LOV.Dark",
  TLR10VsWTInDark = "ADIPO_ASC_TLR10_LOV.Dark - ADIPO_ASC_WT.Dark",
  TLR10VsWTInLight = "ADIPO_ASC_TLR10_LOV.Light - ADIPO_ASC_WT.Light",
  Diff = "(ADIPO_ASC_TLR10_LOV.Light - ADIPO_ASC_TLR10_LOV.Dark) - (ADIPO_ASC_WT.Light - ADIPO_ASC_WT.Dark)"
)

fit <- fit_DEqMS_model(data.imputed, contrast_list)

VarianceBoxplot(fit)

```

### ADIPO ASC Wildtype Light vs. Dark

```{r ADIPO_ASC_WT_WCL_extract_results}

current_contrast <- 1
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "WT")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_WT_Light_vs_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_WT_Light_vs_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_WT_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_WT_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_WT_WCL_volcanoplot}

plot_volcano(res, "ADIPO ASC Wildtype Light vs. Dark Whole Cell Lysate", vp_lfc_limit)

```


### ADIPO ASC TLR10LOV Light vs. Dark

```{r ADIPO_ASC_TLR10_WCL_extract_results}

current_contrast <- 2
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "TLR10_LOV")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_TLR10LOV_Light_vs_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_TLR10LOV_Light_vs_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_TLR10_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_TLR10_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_TLR10_WCL_volcanoplot}

plot_volcano(res, "ADIPO ASC TLR10LOV Light vs. Dark Whole Cell Lysate", vp_lfc_limit)

```


### ADIPO ASC TLR10LOV vs. Wildtype Dark

```{r ADIPO_ASC_Dark_WCL_extract_results}

current_contrast <- 3
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Dark")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_Dark_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_Dark_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_Dark_WCL_volcanoplot}

plot_volcano(res, "ADIPO ASC TLR10LOV vs. Wildtype Dark Whole Cell Lysate", vp_lfc_limit)

```


### ADIPO ASC TLR10LOV vs. Wildtype Light

```{r ADIPO_ASC_Light_WCL_extract_results}

current_contrast <- 4
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Light")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Light_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_WT_vs_TLR10LOV_Light_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_Light_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_Light_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows,
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_Light_WCL_volcanoplot}

plot_volcano(res, "ADIPO ASC TLR10LOV vs. Wildtype Light Whole Cell Lysate", vp_lfc_limit)

```


### ADIPO ASC Interaction (TLR10LOV Light vs. Dark) vs. (Wildtype Light vs. Dark)

```{r ADIPO_ASC_Interaction_WCL_extract_results}

current_contrast <- 5
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, ]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/ADIPO_ASC_Interaction_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/ADIPO_ASC_Interaction_WCL_filtered.csv",
          row.names = FALSE)

```

```{r ADIPO_ASC_Interaction_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r ADIPO_ASC_Interaction_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("ADIPO_ASC_WT" = "lightblue",
                                                       "ADIPO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r ADIPO_ASC_Interaction_WCL_volcanoplot}

plot_volcano(res, "ADIPO ASC Interaction Whole Cell Lysate", vp_lfc_limit)

```


# OSTEO ASC

## Supernatant

```{r OSTEO_ASC_SN_import}

data <- read_DIA_report("data/MS1MS2/20240701_082302_OSTEO_ASC_SN_TLR10_Report.tsv")

data_MS1.df <- assay(data, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
data_MS2.df <- assay(data, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```

```{r OSTEO_ASC_SN_plot_missing_before}

plot_missing(data_MS1.df, data_MS2.df)

```

```{r OSTEO_ASC_SN_filter_na}

keep <- filter_too_many_missing(data_MS1.df, data_MS2.df)

data.filtered <- data[rownames(data) %in% keep,]


data_MS1.filtered.df <- assay(data.filtered, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.filtered.df <- assay(data.filtered, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```


```{r OSTEO_ASC_SN_plot_missing_after}

plot_missing(data_MS1.filtered.df, data_MS2.filtered.df)

```

```{r OSTEO_ASC_SN_log_transform}

data.log2 <- data.filtered
assays(data.log2) %<>% lapply(log2)

data_MS1.log2.df <- assay(data.log2, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.log2.df <- assay(data.log2, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")


plot_intensity_boxplot(data_MS1.log2.df, data_MS2.log2.df)

```

```{r OSTEO_ASC_SN_normalize}

data.norm <- data.log2
assays(data.norm) %<>% lapply(normalize_matrix, "center.median")

data_MS1.norm.df <- assay(data.norm, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.norm.df <- assay(data.norm, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

plot_intensity_boxplot(data_MS1.norm.df, data_MS2.norm.df)

```

```{r OSTEO_ASC_SN_plot_missing_heatmap}

MNAR_MS1 <- get_MNAR(data_MS1.norm.df)
MNAR_MS2 <- get_MNAR(data_MS2.norm.df)
MNAR <- MNAR_MS1 | MNAR_MS2

plot_missing_heatmap(data_MS1.norm.df, data_MS2.norm.df, MNAR, colData(data.norm),
                     colors_columns = list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                      "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                           Condition=c("Light" = "orange",
                                                       "Dark" = "midnightblue")),
                     colors_rows = list(MNAR=c("TRUE"="darkgreen",
                                               "FALSE"="darkmagenta")))

```

```{r OSTEO_ASC_SN_plot_missing_density}

plot_missing_density(data_MS1.norm.df, data_MS2.norm.df)

```

```{r OSTEO_ASC_SN_impute_missing_values}

data.imputed <- data.norm

assays(data.imputed) %<>% lapply(MsCoreUtils::impute_mixed, !MNAR, "knn", "min")

data_MS1.imputed.df <- assay(data.imputed, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.imputed.df <- assay(data.imputed, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
  
```

```{r OSTEO_ASC_SN_plot_PCA}

plot_pca(data.imputed, scale = F)

```

```{r OSTEO_ASC_SN_fit_model}


contrast_list <- c(
  LightVsDarkInWT = "OSTEO_ASC_WT.Light - OSTEO_ASC_WT.Dark", 
  LightVsDarkInTLR10 = "OSTEO_ASC_TLR10_LOV.Light - OSTEO_ASC_TLR10_LOV.Dark",
  TLR10VsWTInDark = "OSTEO_ASC_TLR10_LOV.Dark - OSTEO_ASC_WT.Dark",
  TLR10VsWTInLight = "OSTEO_ASC_TLR10_LOV.Light - OSTEO_ASC_WT.Light",
  Diff = "(OSTEO_ASC_TLR10_LOV.Light - OSTEO_ASC_TLR10_LOV.Dark) - (OSTEO_ASC_WT.Light - OSTEO_ASC_WT.Dark)"
)

fit <- fit_DEqMS_model(data.imputed, contrast_list)

VarianceBoxplot(fit)

```

### OSTEO ASC Wildtype Light vs. Dark

```{r OSTEO_ASC_WT_SN_extract_results}

current_contrast <- 1
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "WT")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_WT_Light_vs_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_WT_Light_vs_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_WT_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_WT_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_WT_SN_volcanoplot}

plot_volcano(res, "OSTEO ASC Wildtype Light vs. Dark Supernatant", vp_lfc_limit)

```


### OSTEO ASC TLR10LOV Light vs. Dark

```{r OSTEO_ASC_TLR10_SN_extract_results}

current_contrast <- 2
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "TLR10_LOV")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_TLR10LOV_Light_vs_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_TLR10LOV_Light_vs_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_TLR10_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_TLR10_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_TLR10_SN_volcanoplot}

plot_volcano(res, "OSTEO ASC TLR10LOV Light vs. Dark Supernatant", vp_lfc_limit)

```


### OSTEO ASC TLR10LOV vs. Wildtype Dark

```{r OSTEO_ASC_Dark_SN_extract_results}

current_contrast <- 3
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Dark")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Dark_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Dark_SN_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_Dark_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_Dark_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_Dark_SN_volcanoplot}

plot_volcano(res, "OSTEO ASC TLR10LOV vs. Wildtype Dark Supernatant", vp_lfc_limit)

```


### OSTEO ASC TLR10LOV vs. Wildtype Light

```{r OSTEO_ASC_Light_SN_extract_results}

current_contrast <- 4
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Light")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Light_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Light_SN_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_Light_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_Light_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows,
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_Light_SN_volcanoplot}

plot_volcano(res, "OSTEO ASC TLR10LOV vs. Wildtype Light Supernatant", vp_lfc_limit)

```


### OSTEO ASC Interaction (TLR10LOV Light vs. Dark) vs. (Wildtype Light vs. Dark)

```{r OSTEO_ASC_Interaction_SN_extract_results}

current_contrast <- 5
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, ]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_Interaction_SN_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_Interaction_SN_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_Interaction_SN_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_Interaction_SN_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, cluster_rows = hm_cluster_rows,
             cluster_cols = hm_cluster_cols, scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_Interaction_SN_volcanoplot}

plot_volcano(res, "OSTEO ASC Interaction Supernatant", vp_lfc_limit)

```


## Whole Cell Lysate

```{r OSTEO_ASC_WCL_import}

data <- read_DIA_report("data/MS1MS2/20240701_082250_OSTEO_ASC_WCL_TLR10_Report.tsv")

data_MS1.df <- assay(data, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
data_MS2.df <- assay(data, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```

```{r OSTEO_ASC_WCL_plot_missing_before}

plot_missing(data_MS1.df, data_MS2.df)

```

```{r OSTEO_ASC_WCL_filter_na}

keep <- filter_too_many_missing(data_MS1.df, data_MS2.df)

data.filtered <- data[rownames(data) %in% keep,]


data_MS1.filtered.df <- assay(data.filtered, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.filtered.df <- assay(data.filtered, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

```


```{r OSTEO_ASC_WCL_plot_missing_after}

plot_missing(data_MS1.filtered.df, data_MS2.filtered.df)

```

```{r OSTEO_ASC_WCL_log_transform}

data.log2 <- data.filtered
assays(data.log2) %<>% lapply(log2)

data_MS1.log2.df <- assay(data.log2, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.log2.df <- assay(data.log2, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")


plot_intensity_boxplot(data_MS1.log2.df, data_MS2.log2.df)

```

```{r OSTEO_ASC_WCL_normalize}

data.norm <- data.log2
assays(data.norm) %<>% lapply(normalize_matrix, "center.median")

data_MS1.norm.df <- assay(data.norm, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.norm.df <- assay(data.norm, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

plot_intensity_boxplot(data_MS1.norm.df, data_MS2.norm.df)

```

```{r OSTEO_ASC_WCL_plot_missing_heatmap}

MNAR_MS1 <- get_MNAR(data_MS1.norm.df)
MNAR_MS2 <- get_MNAR(data_MS2.norm.df)
MNAR <- MNAR_MS1 | MNAR_MS2

plot_missing_heatmap(data_MS1.norm.df, data_MS2.norm.df, MNAR, colData(data.norm),
                     colors_columns = list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                      "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                           Condition=c("Light" = "orange",
                                                       "Dark" = "midnightblue")),
                     colors_rows = list(MNAR=c("TRUE"="darkgreen",
                                               "FALSE"="darkmagenta")))

```

```{r OSTEO_ASC_WCL_plot_missing_density}

plot_missing_density(data_MS1.norm.df, data_MS2.norm.df)

```

```{r OSTEO_ASC_WCL_impute_missing_values}

data.imputed <- data.norm

assays(data.imputed) %<>% lapply(MsCoreUtils::impute_mixed, !MNAR, "knn", "min")

data_MS1.imputed.df <- assay(data.imputed, "MS1") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")

data_MS2.imputed.df <- assay(data.imputed, "MS2") %>%
  as.data.frame() %>%
  rownames_to_column("Gene.Name")
  
```

```{r OSTEO_ASC_WCL_plot_PCA}

plot_pca(data.imputed, scale = F)

```

```{r OSTEO_ASC_WCL_fit_model}


contrast_list <- c(
  LightVsDarkInWT = "OSTEO_ASC_WT.Light - OSTEO_ASC_WT.Dark", 
  LightVsDarkInTLR10 = "OSTEO_ASC_TLR10_LOV.Light - OSTEO_ASC_TLR10_LOV.Dark",
  TLR10VsWTInDark = "OSTEO_ASC_TLR10_LOV.Dark - OSTEO_ASC_WT.Dark",
  TLR10VsWTInLight = "OSTEO_ASC_TLR10_LOV.Light - OSTEO_ASC_WT.Light",
  Diff = "(OSTEO_ASC_TLR10_LOV.Light - OSTEO_ASC_TLR10_LOV.Dark) - (OSTEO_ASC_WT.Light - OSTEO_ASC_WT.Dark)"
)

fit <- fit_DEqMS_model(data.imputed, contrast_list)

VarianceBoxplot(fit)

```

### OSTEO ASC Wildtype Light vs. Dark

```{r OSTEO_ASC_WT_WCL_extract_results}

current_contrast <- 1
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "WT")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_WT_Light_vs_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_WT_Light_vs_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_WT_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_WT_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_WT_WCL_volcanoplot}

plot_volcano(res, "OSTEO ASC Wildtype Light vs. Dark Whole Cell Lysate", vp_lfc_limit)

```


### OSTEO ASC TLR10LOV Light vs. Dark

```{r OSTEO_ASC_TLR10_WCL_extract_results}

current_contrast <- 2
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "TLR10_LOV")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_TLR10LOV_Light_vs_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_TLR10LOV_Light_vs_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_TLR10_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_TLR10_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_TLR10_WCL_volcanoplot}

plot_volcano(res, "OSTEO ASC TLR10LOV Light vs. Dark Whole Cell Lysate", vp_lfc_limit)

```


### OSTEO ASC TLR10LOV vs. Wildtype Dark

```{r OSTEO_ASC_Dark_WCL_extract_results}

current_contrast <- 3
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Dark")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Dark_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Dark_WCL_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_Dark_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_Dark_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_Dark_WCL_volcanoplot}

plot_volcano(res, "OSTEO ASC TLR10LOV vs. Wildtype Dark Whole Cell Lysate", vp_lfc_limit)

```


### OSTEO ASC TLR10LOV vs. Wildtype Light

```{r OSTEO_ASC_Light_WCL_extract_results}

current_contrast <- 4
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, 
                         str_detect(colnames(data.imputed), "Light")]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Light_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_WT_vs_TLR10LOV_Light_WCL_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_Light_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_Light_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows,
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_Light_WCL_volcanoplot}

plot_volcano(res, "OSTEO ASC TLR10LOV vs. Wildtype Light Whole Cell Lysate", vp_lfc_limit)

```


### OSTEO ASC Interaction (TLR10LOV Light vs. Dark) vs. (Wildtype Light vs. Dark)

```{r OSTEO_ASC_Interaction_WCL_extract_results}

current_contrast <- 5
res <- outputResult(fit, coef_col = current_contrast) %>%
  as_tibble() %>%
  mutate(qvalue = qvalue(P.Value)$qvalues,
         sca.qvalue = qvalue(sca.P.Value)$qvalues) %>%
  select(!c(adj.P.Val, sca.adj.pval))

res.sig <- res %>%
  filter(sca.qvalue < 0.05 & abs(logFC) >= 0.58)

qvals.sig <- res.sig$sca.qvalue
names(qvals.sig) <- res.sig$gene

data.sig <- data.imputed[rownames(data.imputed) %in% res.sig$gene, ]

data.sig.df <- assay(data.sig, "MS2") %>% as.data.frame()
  

write.csv(res, file = "results/DEA/OSTEO_ASC_Interaction_WCL_unfiltered.csv",
          row.names = FALSE)

write.csv(res.sig, file = "results/DEA/OSTEO_ASC_Interaction_WCL_filtered.csv",
          row.names = FALSE)

```

```{r OSTEO_ASC_Interaction_WCL_plot_pvalues}

ggplot(data = res, aes(x = sca.P.Value)) + 
  geom_histogram(binwidth = 0.025)

```

```{r OSTEO_ASC_Interaction_WCL_plot_heatmap}

plot_heatmap(data.sig, heatmap.colors, list(Celltype=c("OSTEO_ASC_WT" = "lightblue",
                                                       "OSTEO_ASC_TLR10_LOV" = "lightgreen"),
                                            Condition=c("Light" = "orange",
                                                        "Dark" = "midnightblue")),
             qvalues = qvals.sig, title = "Rel. LogIntensity", max_rows = hm_max_rows, 
             cluster_rows = hm_cluster_rows, cluster_cols = hm_cluster_cols, 
             scale_by_row = hm_scale_by_row)

```

```{r OSTEO_ASC_Interaction_WCL_volcanoplot}

plot_volcano(res, "OSTEO ASC Interaction Whole Cell Lysate", vp_lfc_limit)

```